runtime/interpreter/decode.go

```diff
@@ -1033,8 +1199,8 @@ func (d *DecoderV4) decodeCapability(v interface{}) (CapabilityValue, error) {
 
 	// path
 
-	field2 := encoded[encodedCapabilityValuePathFieldKey]
-	field2Value, err := d.decodeValue(field2, nil)
+	// Decode path at array index encodedCapabilityValuePathFieldKey
```

Comment:
Same here: Could we directly call `decodePath` here? This would avoid the type dispatch in `decodeValue` and the type check below

Reply:
Great idea!